src/elm/clinical.ts

```diff
@@ -217,34 +230,37 @@ export class Concept extends Expression {
   codes: any;
   display?: string;
 
-  constructor(json: any) {
+  constructor(json: ELM.Concept) {
     super(json);
     this.codes = json.code;
     this.display = json.display;
   }
 
   // Define a simple getter to allow type-checking of this class without instanceof
   // and in a way that survives minification (as opposed to checking constructor.name)
-  get isConcept() {
+  get isConcept(): boolean {
     return true;
   }
 
-  toCode(ctx: Context, code: any) {
+  toCode(ctx: Context, code: any): dt.Code {
     const system = ctx.getCodeSystem(code.system.name) || {};
     return new dt.Code(code.code, system.id, code.version, code.display);
   }
 
-  exec(ctx: Context) {
+  exec(ctx: Context): dt.Concept {
     const codes = this.codes.map((code: any) => this.toCode(ctx, code));
     return new dt.Concept(codes, this.display);
   }
 }
 
 export class CalculateAge extends Expression {
-  precision: string;
+  precision: ELM.DateTimePrecision;
```

Comment:
Oooo love that we get the specific literals now rather than having to use general string. Yay!

Reply:
Yeah this is nice. In theory we could also make this an ENUM instead of
```
  type DateTimePrecision =
    | 'Year'
    | 'Month'
    | 'Week'
    | 'Day'
    | 'Hour'
    | 'Minute'
    | 'Second'
    | 'Millisecond';
```
but both probably are fine.